### PR TITLE
Linchpin setup and Adding error handling messages beaker and libvirt

### PR DESCRIPTION
**Pull request by samvarankashyap**
Introducing linchpin setup command which are used to automate linchpin external dependency installations 

command: 
```
linchpin setup [dependency_list]
```
example usage: 
```
# by default linchpin setup installs all dependencies 
# test , docs , libvirt, beaker, selinux etc 
linchpin setup
```
```
# one can always specify set of dependencies
linchpin setup beaker libvirt
```

Further current PR adds debug fail messages to beaker and libvirt when something goes wrong with the provisioning process. It works both in virtualenv.
Some of the dependencies might need  sudo permissions which can either be done using 
```
linchpin setup --ask-sudo-pass 
```
or running linchpin with sudo 
```
sudo linchpin setup
```
or running linchpin under a privileged sudo user 
```
linchpin setup
```

fixes: #419

### CODE REVIEW

File: linchpin/provision/roles/dependencies/tasks/libvirt_deps.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+- name: Install dependencies
+  block:
+    - name: Install package dependencies
+      package:
```
Comment (p3ck) [inline linchpin/provision/roles/dependencies/tasks/libvirt_deps.yml]: Please see https://github.com/ansible/ansible/issues/24581

You should not use with_items with package module.  Pass the list directly to name: in package module.

Comment (14rcole) [reply linchpin/provision/roles/dependencies/tasks/libvirt_deps.yml]: @samvarankashyap fyi this appears in a few places

**Comment by samvarankashyap** [comment]
@14rcole : The reasoning behind the merging error handling and linchpin setup in the current PR is the error messages are suggestion to use linchpin setup before running linchpin provision.
I will try to add the documentation, However I could not find appropriate place for it in docs . Most probably I will try to write something in installation docs as an additional section how to use linchpin setup to install dependencies. 
https://linchpin.readthedocs.io/en/latest/installation.html

**Comment by samvarankashyap** [comment]
@14rcole ??

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**